src/typedefs.rs

```diff
@@ -533,6 +548,21 @@ macro_rules! construct_tag {
         /// # Security:
         /// - __**Avoid using**__ `unprotected_as_bytes()` whenever possible, as it breaks all protections
         /// that the type implements.
+        ///
+        /// - The trait `PartialEq<&'_ [u8]>` is implemented for this type so that users are not tempted
+        /// to call `unprotected_as_bytes` to compare this sensitive value to a byte slice. The trait
+        /// is implemented in such a way that the comparison happens in constant time. Thus, users should
+        /// prefer `SecretType == &[u8]` over `SecretType.unprotected_as_bytes() == &[u8]`.
+        /// Examples are shown below. The examples apply to any type that implements `PartialEq<&'_ [u8]>`.
+        /// ```rust
+        /// use orion::hazardous::mac::hmac::Tag;
+        ///
+        /// // Initialize an arbitrary, 64-byte tag.
+        /// let tag = Tag::from_slice(&[1; 64]).unwrap();
```

Comment:
We want to avoid examples using `.unwrap()` to comply with the [Rust-API guidelines](https://rust-lang-nursery.github.io/api-guidelines/checklist.html). Let the code example end similar to [this one](https://github.com/brycx/orion/blob/8de3292ca6523d6be7e59642ead86a61e05a6958/src/aead.rs#L75) and the `Result` can be propagated with `?`.